gobblin-compliance/src/main/java/gobblin/compliance/HivePurgerExtractor.java

```diff
@@ -0,0 +1,176 @@
+      throws IOException {
+    Partition hiveTablePartition =
+        getHiveTablePartition(state.getWorkunit().getProp(HivePurgerConfigurationKeys.PARTITION_NAME),
+            state.getProperties());
+    Map<String, String> partitionParameters = hiveTablePartition.getTable().getParameters();
+    Preconditions.checkArgument(state.contains(HivePurgerConfigurationKeys.STAGING_DB_KEY),
+        "Missing property " + HivePurgerConfigurationKeys.STAGING_DB_KEY);
+    Preconditions.checkArgument(state.contains(HivePurgerConfigurationKeys.STAGING_DIR_KEY),
+        "Missing property " + HivePurgerConfigurationKeys.STAGING_DIR_KEY);
+    Preconditions.checkArgument(state.contains(HivePurgerConfigurationKeys.COMPLIANCE_IDENTIFIER_KEY),
+        "Missing property " + HivePurgerConfigurationKeys.DATASET_DESCRIPTOR_IDENTIFIER);
+    Preconditions.checkArgument(state.contains(HivePurgerConfigurationKeys.COMPLIANCE_ID_TABLE_KEY),
+        "Missing property " + HivePurgerConfigurationKeys.COMPLIANCE_ID_TABLE_KEY);
+    Preconditions.checkArgument(state.contains(HivePurgerConfigurationKeys.DATASET_DESCRIPTOR_IDENTIFIER),
+        "Missing property " + HivePurgerConfigurationKeys.DATASET_DESCRIPTOR_IDENTIFIER);
+    Preconditions.checkArgument(partitionParameters.containsKey(HivePurgerConfigurationKeys.DATASET_DESCRIPTOR_KEY),
+        "Missing table property " + HivePurgerConfigurationKeys.DATASET_DESCRIPTOR_KEY);
+
+    String datasetDescriptorClass = state.getProp(HivePurgerConfigurationKeys.DATASET_DESCRIPTOR_CLASS,
+        HivePurgerConfigurationKeys.DEFAULT_DATASET_DESCRIPTOR_CLASS);
+    DatasetDescriptor descriptor = GobblinConstructorUtils
+        .invokeConstructor(DatasetDescriptor.class, datasetDescriptorClass,
+            partitionParameters.get(HivePurgerConfigurationKeys.DATASET_DESCRIPTOR_KEY),
+            state.getProp(HivePurgerConfigurationKeys.DATASET_DESCRIPTOR_IDENTIFIER));
+
+    Boolean commit =
+        state.getPropAsBoolean(HivePurgerConfigurationKeys.COMMIT_KEY, HivePurgerConfigurationKeys.DEFAULT_COMMIT);
+    String stagingDir = state.getProp(HivePurgerConfigurationKeys.STAGING_DIR_KEY);
+    String stagingDb = state.getProp(HivePurgerConfigurationKeys.STAGING_DB_KEY);
+    String complianceIdentifier = state.getProp(HivePurgerConfigurationKeys.COMPLIANCE_IDENTIFIER_KEY);
+    String complianceIdTable = state.getProp(HivePurgerConfigurationKeys.COMPLIANCE_ID_TABLE_KEY);
+
+    return new ComplianceRecord.ComplianceRecordBuilder().withComplianceIdentifier(complianceIdentifier)
+        .withComplianceIdTable(complianceIdTable).withCommit(commit).withStagingDb(stagingDb).withStagingDir(stagingDir)
+        .withDatasetDescriptor(descriptor).withHivePartition(hiveTablePartition).withJobId(state.getJobState().getId())
+        .build();
+  }
+
+  /**
+   * Get {@link Partition} from the partition name.
+   * @throws IOException
+   */
+  protected Partition getHiveTablePartition(String partitionName, Properties properties)
```

Comment:
Any reason the state object is converted to a Properties collection? It looks unnecessary.